src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/keycloak-client/main.tf

```diff
@@ -67,29 +67,54 @@ data "keycloak_realm" "master" {
   realm = "nebari"
 }
 
-data "keycloak_openid_client" "realm_management" {
-  realm_id  = var.realm_id
-  client_id = "realm-management"
-}
 
-data "keycloak_role" "main-service" {
-  for_each = toset(var.service-account-roles)
+# Get client data for each service account client
+data "keycloak_openid_client" "service_clients" {
```

Comment:
Before we only allowed service accounts to get roles from the realm-management client.  This PR allows us to set roles by any client.  This functionality was needed to be able to set the allow-app-sharing-role on the jupyterhub service account.